app/src/controllers/testCase/sagas.ts

```diff
@@ -52,12 +57,79 @@ function* getTestCases(action: GetTestCasesAction) {
 }
 
 function* getFolders() {
+  const projectKey = (yield select(projectKeySelector)) as string;
+
+  if (!projectKey) {
+    return;
+  }
+
   try {
-    const projectKey = (yield select(projectKeySelector)) as string;
-    const folders = (yield call(fetch, URLS.folder(projectKey))) as { content: Folder };
-    yield put(setFoldersAction(folders.content));
+    yield put({
+      type: FETCH_START,
+      payload: { projectKey },
+      meta: { namespace: NAMESPACE },
+    });
+
+    try {
+      const pageSize = 20;
+      let pageNumber = 1;
+      let totalPages = 1;
+      const aggregated: Folder[] = [];
+
+      type FoldersPage = { content: Folder[]; page?: { totalPages?: number } };
+      const isFoldersPage = (value: unknown): value is FoldersPage => {
+        if (!value || typeof value !== 'object') return false;
+        const v = value as { content?: unknown; page?: unknown };
+        return Array.isArray(v.content);
+      };
+
+      while (pageNumber <= totalPages) {
+        const responseUnknown: unknown = yield call(
```

Comment:
1. Do I undrerstand it right that if at least one request of batch fails it will end up in catch block? 
2. How will UI look exactly in this case?

Reply:
@ViktorSoroka07 I bet we need to define some expectations, until then it is undefined behaviour

Reply:
I just want to understand what will happen now. The notification text I think you can clarify with Teo and Anatoliy. With regard to UI I would expect for now the empty state

Reply:
added notification.
<img width="1086" height="1078" alt="image" src="https://github.com/user-attachments/assets/d7fc4eb4-dc3a-4156-8079-ff7791836ff0" />